projects/rocr-runtime/runtime/hsa-runtime/image/resource.h

```diff
@@ -202,6 +203,68 @@ typedef struct Sampler {
   hsa_ext_sampler_descriptor_v2_t desc;
 } Sampler;
 
+/// @brief Structure representing a mipmapped image array.
+typedef struct MipmappedArray : public Image {
+private:
+  MipmappedArray() {
+    component.handle = 0;
+    data = NULL;
+    size = 0;
+    std::memset(srd, 0, sizeof(srd));
+    std::memset(&desc, 0, sizeof(desc));
+    permission = HSA_ACCESS_PERMISSION_RO;
+    num_levels = 0;
+    flags = 0;
+    std::memset(&addr_output, 0, sizeof(addr_output));
+    row_pitch = slice_pitch = 0;
+    tile_mode = LINEAR;
+  }
+
+~MipmappedArray() {
```

Comment:
should   ~Image() {} be virtual?
Looks like Image::Destroy(const Image* image) need that to destroy MipmappedArray